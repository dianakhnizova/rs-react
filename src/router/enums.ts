export enum PagePath {
  root = '/',
  aboutPage = '/about',
}
