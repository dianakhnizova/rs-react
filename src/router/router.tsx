import { createBrowserRouter } from 'react-router-dom';
import { PagePath } from './enums';
import { AboutPage } from '@/pages/about-page/AboutPage';
import { Root } from './root/root';
import { MainPage } from '@/pages/main-page/MainPage';
import { NotFoundPage } from '@/pages/not-found-page/NotFoundPage';
import { BookDetailPage } from '@/pages/book-detail-page/BookDetailPage';

export const routes = [
  {
    path: PagePath.root,
    Component: Root,
    children: [
      { index: true, Component: MainPage },
      { path: PagePath.aboutPage, Component: AboutPage },
      { path: PagePath.notFound, Component: NotFoundPage },
      {
        path: PagePath.mainPage,
        Component: MainPage,
        children: [
          {
            path: PagePath.bookDetailPage,
            Component: BookDetailPage,
          },
        ],
      },
    ],
  },
  {
    path: PagePath.notFound,
    Component: NotFoundPage,
  },
];

export const router = createBrowserRouter(routes);
