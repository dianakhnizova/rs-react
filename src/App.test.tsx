import { render, screen } from '@testing-library/react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { describe, it } from 'vitest';
import { routes } from './router/router';

describe('App routing', () => {
  it('renders MainPage on default route', () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ['/'],
    });

    render(<RouterProvider router={router} />);

    expect(screen.getByTestId('main-page')).toBeInTheDocument();
  });

  it('renders AboutPage on /about route', () => {
    const router = createMemoryRouter(routes, {
      initialEntries: ['/about'],
    });

    render(<RouterProvider router={router} />);

    expect(screen.getByTestId('about-page')).toBeInTheDocument();
  });
});
