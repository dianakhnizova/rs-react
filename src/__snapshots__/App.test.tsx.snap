// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches snapshot 1`] = `
<div>
  <header
    class="_container_6f96bd"
  >
    <h1
      class="_title_6f96bd"
    >
      Your Library
    </h1>
  </header>
  <main
    class="_container_81d5d0"
  >
    <div
      class="_overlay_734bb7"
      data-testid="popup-overlay"
    />
    <div
      class="_container_734bb7"
    >
      <div
        class="_container_9a8cdc"
      >
        <div
          class="_imgContainer_9a8cdc"
        >
          <img
            alt="Spinner"
            class="_image_9a8cdc"
            data-testid="spinner-img"
            src="/src/assets/book.png"
          />
        </div>
        <h2
          class="_loadingTitle_9a8cdc"
        >
          Loading...
        </h2>
      </div>
      <button
        class="_button_89f329"
      >
        Close
      </button>
    </div>
    <form
      class="_container_eefda4"
    >
      <input
        class="_input_eefda4"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="_button_89f329 _button_eefda4"
      >
        Search
      </button>
    </form>
    <div
      class="_container_3f70f2"
    >
      <div
        class="_productsHeaderContainer_01c874"
      >
        <p
          class="_titleName_01c874"
        >
          Name
        </p>
        <p
          class="_titleDescription_01c874"
        >
          Description
        </p>
        <p
          class="_titleImage_01c874"
        >
          Image
        </p>
      </div>
      <div
        class="_gridDivider_3f70f2"
      />
      <p
        class="_title_d3cc1b"
      >
        Not books for you
      </p>
    </div>
    <button
      class="_button_89f329 _button_81d5d0"
    >
      Error
    </button>
  </main>
  <footer
    class="_container_50967c"
  >
    <p>
      Â© Diana Khnizova, Inc. All rights reserved.
    </p>
  </footer>
</div>
`;
